src/components/numberpad/NumberPad.tsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable functional/immutable-data */
+import React, { ComponentProps, useRef } from "react";
+import { View } from "react-native";
+import { BiometricsValidType } from "../../utils/types";
+import { IONumberPadButtonStyles, IOStyles } from "../../core";
+import { VSpacer } from "../spacer";
+import { IconButton } from "../buttons";
+import { IOIconSizeScale, IOIcons } from "../icons";
+import { NumberButton } from "./NumberButton";
+
+type BiometricAuthProps =
+  | {
+      biometricType: BiometricsValidType;
+      onBiometricPress: () => void;
+      biometricAccessibilityLabel: string;
+    }
+  | {
+      biometricType?: never;
+      onBiometricPress?: never;
+      biometricAccessibilityLabel?: never;
+    };
+
+type NumberPadProps = {
+  onValueChange: (value: string) => void;
+  variant: ComponentProps<typeof NumberButton>["variant"];
+  deleteAccessibilityLabel: string;
+} & BiometricAuthProps;
+
+const mapIconSpecByBiometric: Record<
+  BiometricsValidType,
+  { icon: IOIcons; size: IOIconSizeScale }
+> = {
+  FACE_ID: { icon: "biomFaceID", size: 32 },
+  TOUCH_ID: { icon: "fingerprint", size: 24 },
+  BIOMETRICS: { icon: "fingerprint", size: 24 }
+};
+
+const ButtonWrapper = ({ children }: { children: React.ReactNode }) => (
+  <View
+    style={[
+      IONumberPadButtonStyles.buttonSize,
+      IOStyles.alignCenter,
+      IOStyles.centerJustified
+    ]}
+  >
+    {children}
+  </View>
+);
+export const NumberPad = ({
+  variant = "dark",
+  onValueChange,
+  biometricType,
+  onBiometricPress,
+  biometricAccessibilityLabel,
+  deleteAccessibilityLabel
+}: NumberPadProps) => {
+  const numberPadValue = useRef<string>("");
+
+  const numberPadPress = (number: number) => {
+    numberPadValue.current = `${numberPadValue.current}${number}`;
+    onValueChange(numberPadValue.current);
+  };
+
+  const onDeletePress = () => {
+    numberPadValue.current = numberPadValue.current.slice(0, -1);
+    onValueChange(numberPadValue.current);
+  };
+
+  type ButtonType = "biometric" | "delete";
+
+  const RowButtons = ({
```

Comment:
Could this component be extracted as a standalone, reusable component, separate from the NumberPad?

Reply:
Actually this should be only a local component avoiding code duplication to repeat button lines. Do you see any other possible usage of the component? 😄